Rock.Messaging/MQ/MessagingScenarioFactory.cs

```diff
@@ -59,17 +67,38 @@ private static bool TryGetFactoryFromConfig(out IMessagingScenarioFactory factor
         {
             try
             {
-                var rockMessagingConfiguration = (IRockMessagingConfiguration)ConfigurationManager.GetSection("rock.messaging");
-                factory = rockMessagingConfiguration.MessagingScenarioFactory;
+
+                factory = BuildFactory();
+
                 return true;
             }
-            catch (Exception)
+            catch (Exception e)
             {
+                Trace.TraceError($"[Rock.Messaging] - [Messaging Scenario Factory] - TryGet Config failed with {e.Message}");
+
                 factory = null;
                 return false;
             }
         }
 
+        internal static IMessagingScenarioFactory BuildFactory()
+        {
+#if ROCKLIB
+            var messagingSection = Config.Root.GetSection("RockLib.Messaging").Get<ScenarioFactorySection>();
```

Comment:
do we have concerns that the GetSection() could be null which throws an error?

Reply:
I don't think so - the call to `BuildFactory` is wrapped in a try/catch.

Reply:
Agreed.  The error is going to wrapped up further up.